src/GPE.jl

```diff
@@ -134,10 +157,10 @@ end
 
 Initialise the marginal log-likelihood of Gaussian process `gp`.
 """
-function initialise_mll!(gp::GPE)
+function initialise_mll!(gp::GPE{X,Y,M,K,P,D}) where {X,Y,M,K,P,D}
     μ = mean(gp.mean,gp.x)
     Σ = cov(gp.kernel, gp.x, gp.data)
-    gp.cK = PDMat(Σ + (exp(2*gp.logNoise) + 1e-5)*I)
+    gp.cK = P.name.wrapper(Σ + (exp(2*gp.logNoise) + 1e-5)*I)
```

Comment:
I would prefer to use multiple dispatch and have separate functions for PDMat and ElasticMat for initialising the covariance matrix.